### PR TITLE
[fix](memory) Fix LRU cache frequent prune

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Memory GC and periodic thread will prune LRU cache, and will hold cache lock when prune.
If the cache is frequently written, frequent prune cache will affect performance and block memory GC.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/runtime/memory/cache_manager.h [hunk 1]
```diff
@@ -54,9 +55,29 @@ class CacheManager {
 
     void clear_once(CachePolicy::CacheType type);
 
+    template <bool prune_stale>
+    bool need_prune() {
```
Comment (yiguolei) [inline be/src/runtime/memory/cache_manager.h:59]: 这里不要用模板参数，直接用函数的参数就行了

File: be/src/olap/olap_server.cpp [hunk 2]
```diff
@@ -287,6 +286,16 @@ void StorageEngine::_cache_clean_callback() {
         } else {
             _clear_segment_cache = false;
         }
+        if (config::disable_storage_page_cache) {
```
Comment (yiguolei) [inline be/src/olap/olap_server.cpp:289]: 为啥是在disable page cache的时候去clear 呢？

File: be/src/runtime/memory/cache_manager.h (hunk 1 above)
Comment (xinyiZzz) [reply be/src/runtime/memory/cache_manager.h]: done

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by wangbo — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**